Update loki-operator and loki

vyzigold opened the pull request:
Update the loki-operator to the most recent version and also update loki to version 2.4.2.

csibbitt commented on build/stf-run-ci/defaults/main.yml:
```diff
@@ -20,21 +20,21 @@ __internal_registry_path: image-registry.openshift-image-registry.svc:5000
 # Set a default commit hash to clone for loki-operator to freeze
 # the operator developement.
-loki_operator_branch: 700e275
+loki_operator_branch: b8e9973
```
spooky magic numbers

vyzigold replied:
the loki-operator is under development and there isn't a stable version released yet, so this is a commit hash

csibbitt reviewed (approved):
LGTM

leifmadsen commented on build/stf-run-ci/defaults/main.yml:
```diff
@@ -20,21 +20,21 @@ __internal_registry_path: image-registry.openshift-image-registry.svc:5000
 prometheus_webhook_snmp_image_tag: latest
 loki_operator_image_tag: latest
-new_operator_sdk_version: v1.5.0
+new_operator_sdk_version: v1.11.0
```
Thanks Chris, yea this is because Loki Operator is built with a newer version of operator-sdk than what we use for our components. This was actually one of the issues I had to deal with as part of the operator-sdk migration.

This would get dealt with should we decide to continue with the effort in migrating to operator-sdk 1.16 (or otherwise). Ideally Loki would exist somewhere that we could use. I'm thinking that instead of _building_ Loki Operator directly into our test framework, we should actually consider just making it available in a CatalogSource / index image like what is available for Prometheus, ECK etc, and maybe just pull all of this logic out entirely.

We don't maintain Loki and are unlikely to be the ones that will productize it. Keeping this kind of build logic in our test framework is, in my opinion, just making our test framework more complicated for no reason.

I might look at how hard it would be to just create the bundle and index image now that I've got some > 1.0 operator-sdk experience. I think it will be relatively straight forward.

No changes required for this PR. I might see if I can hack something together quickly and pull out some of the custom logic in our test framework to clean it up.

vyzigold commented on build/stf-run-ci/tasks/setup_stf_local_build.yml:
```diff
@@ -164,7 +170,7 @@
   when:
     - not __deploy_minio_enabled | bool
-    - __service_telemetry_observability_strategy == "use_community"
+    - not __service_telemetry_observability_strategy == "use_community"
```
Well, this is quite confusing, but minio is getting deployed by default, when we deploy Loki and this is to **remove** the deployment. In other words, what is happening here with this PR: If observability_strategy = none, minio deployment gets removed, so it doesn't get removed. I hope I'm right, because it also took me a minute.

leifmadsen reviewed (approved):
LGTM

leifmadsen commented:
@vyzigold while CI is down, if you run the `ansible-playbook build/run-ci.yml` and `./tests/smoketest/smoketest.sh` and provide the passing output we can merge this down.

vyzigold commented:
Hi @leifmadsen I finally got to this, sorry it took so long. I rebased this on top of current master and added 2 commits with some minor fixes.
The outputs of the commands you wanted me to run: 
ansible: https://paste.opendev.org/show/bmDrYxuNKXiw1X5aIMpR/
smoketest: https://paste.opendev.org/show/bhLPut40iCw9ChKoFRQT/

There isn't cloning and building in the ansible paste, because it looks like the ansible times out while waiting for some of the builds to finish, so I usually need to wait for that build to finish, comment whatever is already built and restart the playbook. This always happened to me and so it isn't an issue of this PR. Otherwise everything finished successfully
